src/components/Wallet/WalletGenerator.jsx

```diff
@@ -66,9 +84,11 @@ class WalletGenerator extends React.Component {
   }
 
   componentDidMount() {
-    const { setIsWallet, refreshNodes } = this.props;
+    const { setIsWallet, refreshNodes, common: {nodesLoaded} } = this.props;
+    this.throttleTestConnection = debounce(args => this.testConnection(args), 500, { trailing: true, leading: false })
```

Comment:
We're using this to automatically test for a valid connection if the setting has been updated. Previously there was no way to update private client settings even though this is necessary for when you import a config since the password isn't saved. This will test, but debounce it so you're not overloading with queries after every key press.